kafka-connect-aws-s3/README-source.md

```diff
@@ -0,0 +1,183 @@
+
+    STOREAS `Text`
+    
+It may be required to use the additional configuration options for this connector to ensure that the value is presented as a String.
+
+    value.converter=org.apache.kafka.connect.storage.StringConverter
+    
+
+#### CSV Input Format Configuration
+
+This reads CSV files written to S3 into an Avro struct to be written back to the Kafka source.
+There are 2 options for CSV format, to read CSV files with the column headers or without.
+
+    STOREAS `CSV_WithHeaders`
+    STOREAS `CSV`
+    
+If the headers are included then the Avro message will use these as the field names.
+
+If the headers are not included then field names will be created using the prefix 'col'. (eg. col1, col2... col<sup>n</sup>)
+
+#### Byte(Binary) Input Format Configuration
+
+Bytes can be read back in from S3 and back into message keys/values, depending on how the data was written to the source.
+
+This can be used for reading back in a messages containing binary data that were written out using the s3 source, or alternatively reading binary files into chunks to be loaded onto a Kafka queue.
+
+Please see the KCQL options available and the results of these configurations:
+
+
+| Option                     | KCQL Configuration                   | Records read from file as  |
+|----------------------------|--------------------------------------|----------------------------|
+| Key and Value (With Sizes) | STOREAS `Bytes_KeyAndValueWithSizes` | Long Long Bytes Bytes      | 
+| Key (With Size)            | STOREAS `Bytes_KeyWithSize`          | Long Bytes                 |
+| Value (With Size)          | STOREAS `Bytes_ValueWithSize`        | Long Bytes                 |
+| Key Only                   | STOREAS `Bytes_KeyOnly`              | Bytes                      |
+| Value Only                 | STOREAS `Bytes_ValueOnly`            | Bytes                      |
+
+
+Using the "With Sizes" options the Source assumes that the files will contain one or two (depending on configuration) 8-byte chunks of data at the start of the file instructing how many bytes to read for the content.
+
+In order to ensure the message is passed through as bytes it may be necessary to set the additional configuration options
+
+    value.converter=org.apache.kafka.connect.converters.ByteArrayConverter
+    key.converter=org.apache.kafka.connect.converters.ByteArrayConverter
+
+Only when using KeyOnly and ValueOnly mode, the size of the chunks of data can be controlled using the additional KCQL option:
+
+    BATCH EQUAL batch_size
+    
+For example, if you want a file to be split into ~0.5 MB chunks then you can configure it using this syntax:
+
+    insert into $TOPIC_NAME select * from $BUCKET_NAME:$PREFIX_NAME BATCH=500000 STOREAS `Bytes_ValueOnly`
+
+
+## Partitioning Options
+
+When using the S3 Source it is required to specify partitioning options so that the connector understands the layout of the file system.
```

Comment:
How does this work if we have CSV ina custom partition with 3 fields?
If there's a custom partitioner it means the fields and values need to be appended.  Do we do that?

Reply:
The regex largely ignores the partition names and just focuses on the last part of the path.

To construct the objects the configuration is required.

It might be possible to refactor so as to reduce the configuration for the source, but for now this works.